### ISSUE TITLE
python client don't work when log server requires SNI

### ISSUE THREAD

**Issue by droopycom** [google/certificate-transparency#1089]
When trying to run some of the sample python clients or tools against log servers that require SNI, the tools will fail.

One such server is the certly.io log. 
Trying to run the simple_scan.py tool on that server will fail because the SSL certificate returned will be for certly.io instead of log.certly.io.

Google logs servers don't have this issue - I assume they server a valid cert for the logs domain even when SNI is not specified.

The root issue is that httplib2 does not appear to support SNI (https://code.google.com/p/httplib2/issues/detail?id=233)

This was tested on OS X, with current master (250672b5aef3666edbdfc9a75b95a09e7a57ed08)

**Comment by eranmes** [comment]
Which Python version have you got?
Python 2.7.9 and up supports SNI, and if you can't use that version then a work-around is using PyOpenSSL + ndg_httpsclient + pyasn1.

The actual output of an attempted run would help.

**Comment by droopycom** [comment]
Python 2.7.11. The one installed by homebrew on OSX.

The problem is not that python does not support SNI, as you mention that has been fixed since 2.7.9, but httplib2 needs to make some change as well apparently. (At least according to https://code.google.com/p/httplib2/issues/detail?id=233)

I worked around the issue by hacking the code to use requests (http://docs.python-requests.org/en/latest/) instead of httplib2. Doing this was interesting because the code around httplib2 is actually creating structure that looks like the one requests use, as indicated by comments in the code.

Attached is a python transcript of manually doing the a get-sth request to the certly log with httplib2.

[httplib-sni-issue.txt](https://github.com/google/certificate-transparency/files/105584/httplib-sni-issue.txt)
